flutter/news/lib/features/frontpage/data/repositories/articles_repository.dart

```diff
@@ -1,43 +1,26 @@
-import 'package:dartz/dartz.dart';
-import 'package:news/core/error/exceptions.dart';
+import 'package:news/core/result.dart';
 
-import '../../../../core/error/failures.dart';
-import '../../../../core/network/network_info.dart';
 import '../../domain/entities/article.dart';
 import '../datasource/articles_data_source.dart';
 import '../datasource/articles_local_data_source.dart';
 
 class ArticlesRepository {
   final ArticlesLocalDataSource localDataSource;
   final ArticlesRemoteDataSource remoteDataSource;
-  final NetworkInfo networkInfo;
```

Comment:
Not required, in case we need it it will be on the `RemoteDataSource` once implemented, and we make it fail fast if there is no connectivity by returning `Failure`. But the repository should be agnostic of it